src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -1661,63 +1833,59 @@ int SketchObject::fillet(int GeoId1, int GeoId2,
 
         // create arc from known parameters and lines
         int filletId;
-        Part::GeomArcOfCircle *arc = Part::createFilletGeometry(lineSeg1, lineSeg2, filletCenter, radius);
-        if (arc) {
-            // calculate intersection and distances before we invalidate lineSeg1 and lineSeg2
-            if (!find2DLinesIntersection(lineSeg1, lineSeg2, intersection)) {
-                delete arc;
-                return -1;
-            }
-            dist1.ProjectToLine(arc->getStartPoint(/*emulateCCW=*/true)-intersection, dir1);
-            dist2.ProjectToLine(arc->getStartPoint(/*emulateCCW=*/true)-intersection, dir2);
-            Part::Geometry *newgeo = arc;
-            filletId = addGeometry(newgeo);
-            if (filletId < 0) {
-                delete arc;
-                return -1;
-            }
-        }
-        else
+        std::unique_ptr<Part::GeomArcOfCircle> arc(
+            Part::createFilletGeometry(lineSeg1, lineSeg2, filletCenter, radius));
+        if (!arc) return -1;
+
+        // calculate intersection and distances before we invalidate lineSeg1 and lineSeg2
+        if (!find2DLinesIntersection(lineSeg1, lineSeg2, intersection)) {
             return -1;
+        }
+
+        dist1.ProjectToLine(arc->getStartPoint(/*emulateCCW=*/true)-intersection, dir1);
+        dist2.ProjectToLine(arc->getStartPoint(/*emulateCCW=*/true)-intersection, dir2);
+        Part::Geometry *newgeo = arc.get();
+        filletId = addGeometry(newgeo);
 
         if (trim) {
             PointPos PosId1 = (filletCenter-intersection)*dir1 > 0 ? start : end;
             PointPos PosId2 = (filletCenter-intersection)*dir2 > 0 ? start : end;
 
-            delConstraintOnPoint(GeoId1, PosId1, false);
-            delConstraintOnPoint(GeoId2, PosId2, false);
-            Sketcher::Constraint *tangent1 = new Sketcher::Constraint();
-            Sketcher::Constraint *tangent2 = new Sketcher::Constraint();
+            if (createCorner) {
+                transferFilletConstraints(GeoId1, PosId1, GeoId2, PosId2);
+            } else {
+                delConstraintOnPoint(GeoId1, PosId1, false);
+                delConstraintOnPoint(GeoId2, PosId2, false);
+            }
 
-            tangent1->Type = Sketcher::Tangent;
-            tangent1->First = GeoId1;
-            tangent1->FirstPos = PosId1;
-            tangent1->Second = filletId;
+            Sketcher::Constraint tangent1, tangent2;

```

Comment:
From a programatic point of view it is ok to use a local variable that will be cloned in addConstraint just before going out of scope. 

In another PR:
https://github.com/FreeCAD/FreeCAD/pull/4330/files#diff-6aa26366ba1f3fa49e7ec312a1b88b41c32ac91df22366ecdc5349bf6dc36619R126

I will be adding: int addConstraint(std::unique_ptr<Constraint> constraint);

this version does not clone the pointer internally, then this could be done by constructing a smart pointer (make_unique) and moving it directly.

However, I only say this for information. There is nothing you can do at this moment and your code will still work afterwards.


I think I have added support for moving a smartpointer in another PR that is currently pending. Eventually, this

Reply:
Thanks for the heads up.